app/src/migration/java/org/mozilla/fenix/MigratingFenixApplication.kt

```diff
@@ -38,6 +39,10 @@ class MigratingFenixApplication : FenixApplication() {
 
         // The rest of the migrations can happen now.
         migrator.startMigrationIfNeeded(migrationStore, MigrationService::class.java)
+
+        // Start migration UI
+        val intent = Intent(this, MigrationProgressActivity::class.java)
+        startActivity(intent)
```

Comment:
This part is untested, but I believe this is the place to start the UI.

Reply:
* This does always start the migration UI? Wouldn't we need to check the state first?
 * As-is this doesn't work, I think:
   * From an application context you can't launch an activity without at least adding the FLAG_ACTIVITY_NEW_TASK flag.
   * For application starts without UI (e.g. some random background task) we wouldn't want to suddenly pop up the migration UI.

Reply:
I wonder if we have to launch it (and shortcut) `HomeActivity`? I assume that's where regular app launches as well as intents (after the migration processor ignored them) end up?

Reply:
I think you're right. I just nicked this bit in on the end, and wanted to get feedback if this was the right location. I'll look into this more now.